src/v/redpanda/admin_server.cc

```diff
@@ -4819,3 +4856,76 @@ admin_server::restart_service_handler(std::unique_ptr<ss::http::request> req) {
     co_await restart_redpanda_service(*service);
     co_return ss::json::json_return_type(ss::json::json_void());
 }
+
+debug_bundle_params make_debug_bundle_params(const ss::http::request& req) {
+    // NOTE: Here, we sanitize for ss::experimental::process (i.e., posix_spawn)
+    // instead of RPK
+    const std::regex bundle_regex("[a-zA-Z0-9\\-\\.]+");
+    auto maybe_sanitize_debug_bundle_param =
+      [&bundle_regex](ss::sstring p) -> std::optional<ss::sstring> {
+        if (p.empty()) {
+            return std::nullopt;
+        }
+
+        validate_utf8(p);
+        if (!std::regex_match(p.c_str(), bundle_regex)) {
+            throw ss::httpd::server_error_exception(fmt::format(
+              "Query param contains invalid characters: param {}", p));
+        }
+
+        return std::make_optional(p);
+    };
+
+    debug_bundle_params bundle_params;
+    bundle_params.logs_since = maybe_sanitize_debug_bundle_param(
+      req.get_query_param("logs-since"));
+    bundle_params.logs_until = maybe_sanitize_debug_bundle_param(
+      req.get_query_param("logs-until"));
+    bundle_params.logs_size_limit = maybe_sanitize_debug_bundle_param(
+      req.get_query_param("logs-size-limit"));
+    bundle_params.metrics_interval = maybe_sanitize_debug_bundle_param(
+      req.get_query_param("metrics-interval"));
+    return bundle_params;
+}
+
+ss::future<std::unique_ptr<ss::http::reply>>
+admin_server::start_debug_bundle_handler(
+  std::unique_ptr<ss::http::request> req,
+  std::unique_ptr<ss::http::reply> rep,
+  const request_auth_result& auth_state) {
+    vlog(logger.info, "Start creating a debug bundle");
+    auto bundle_params = make_debug_bundle_params(*req);
+    co_await _debug_bundle.local().start_creating_bundle(
+      auth_state, std::move(bundle_params));
+    rep->set_status(ss::http::reply::status_type::accepted);
+    co_return std::move(rep);
+}
+
+ss::future<ss::json::json_return_type>
+admin_server::check_debug_bundle_status_handler(
+  std::unique_ptr<ss::http::request> req) {
+    vlog(logger.info, "Getting debug bundle status...");
+    auto bundle_status = co_await _debug_bundle.local().get_status();
+    auto bundle_name = req->param["filename"];
+    detail::throw_if_bundle_dne(
+      bundle_status, _debug_bundle.local().get_write_dir(), bundle_name);
+    co_return ss::json::json_return_type(ss::json::json_void());
+}
```

Comment:
I don't know if this is an issue or not, but since we are throwing exceptions to set the response type, we get a lot of WARN level logs out of the `admin_api_server`:

`WARN  2023-05-18 12:30:36,316 [shard  0] admin_api_server - admin_server.cc:555 - [_anonymous] exception intercepted - url: [http://localhost:9644/v1/debug/bundle/files/debug-bundle.zip] http_return_status[410] reason - seastar::httpd::base_exception (The debug bundle is not running or it is not on disk)`

Pointing this out not as a problem, but something that may be 'annoying' to deal with?

Reply:
Sure, a different solution would be to set the status directly in the reply and write the appropriate message in the response body.